crates/runtime/src/table.rs

```diff
@@ -580,7 +583,11 @@ impl Drop for Table {
     fn drop(&mut self) {
         let ty = self.element_type();
 
-        // funcref tables can skip this
+        // `funcref` tables don't need drops.
+        //
+        // This is an irrefutable pattern when the `gc` cargo feature is not
+        // enabled.
+        #[allow(irrefutable_let_patterns)]
```

Comment:
This is fine as-is but one trick I've done in the past is at the root of the crate:

```rust
#![cfg_attr(not(feature = "gc"), allow(irrefutable_let_patterns))]
```

or more often `unused_imports` or similar.